### ISSUE TITLE
RxCommandStream inputObservable not closed on dispose

### ISSUE THREAD

**Issue by kuhnroyal** [fluttercommunity/rx_command#28]
The input observable for `RxCommandStream` [here](https://github.com/fluttercommunity/rx_command/blob/master/lib/rx_command.dart#L644) is not closed when the command is disposed.

I am wondering if this should be handled by the RxCommand itself or if I should manually add some `takeUntil` logic, which seems like lots of boilerplate.

**Comment by escamoteur** [comment]
Good catch, I really wished we had destructors.
Have to think about it. 
Do you create and dispose Commands a lot while your App is running?

**Comment by kuhnroyal** [comment]
Well, we create commands for every manager/BLoC and as you move through the app from screen to screen some managers get disposed and rebuild.

**Comment by escamoteur** [comment]
Could you live with a dispose method on commands?
In RxVMS most Managers stay alive as long as the app runs. 
BTW if you are using get_it, it now supports disposal of registered objects.

**Comment by escamoteur** [comment]
Sorry that I didn't have time yet.
For the Listview I would only use one command in a manager that takes the item I'd as parameter

**Comment by kuhnroyal** [comment]
But then they would have to fetch their data sequentially as one command can only track executing state for one ID parameter.

**Comment by escamoteur** [comment]
Hi,
I m back at home Thursday. I m in Bonn Germany, so perhaps its the easiest thing to have a short Skype?

**Comment by escamoteur** [comment]
@kuhnroyal Interested in a short chat on this topic?

**Comment by escamoteur** [comment]
I had now a closer look, sorry for the delay.

```Dart
    var inputObservable = Observable(_observableProvider(param))
        .handleError((error) {
          thrownException = error;
        })
        .doOnData((result) => _resultsSubject.add(result))
        .map((result) {
          lastResult = result;
          return CommandResult(result, null, true);
        });

    _commandResultsSubject.addStream(inputObservable).then((_) {
      if (thrownException != null) {
        if (throwExceptions) {
          _resultsSubject.addError(thrownException);
```

You cannot close an observable, you only can close a StreamController or cancel a subscription.
`inputObservable` get's listened to inside `addStream` without storing the subscription anywhere so I can't cancel it from there but I don't think this is a problem.
Otherwise the Dart team would have dealt with this differently.

I also changed the sequence of the dispose method.

**Comment by escamoteur** [comment]
Fixed in V4.3.3
